### PR TITLE
phaser CAN crop

### PR DESCRIPTION
**Pull request by WilliamHarmonMS**
@joshlory or @Hamms 

Phaser can crop the destruction animation! Hurrah!

This crops the destruction overlay to the top 40x40 and then scoots it down 20 pixels so that it lines up over the index the player is destroying so that it can appropriately cover "flat" objects that need an overlay (basically only pressurePlates and rails).

this fixes [issue397](https://github.com/code-dot-org/craft/issues/397)

### CODE REVIEW

File: src/js/game/LevelMVC/LevelView.js [hunk 1]
```diff
@@ -1150,13 +1150,18 @@ module.exports = class LevelView {
...
+    if (LevelBlock.skipsDestructionOverlay(blockType)) {
       // "flat" blocks are by definition not cube shaped and so shouldn't accept
       // the cube-shaped destroy overlay animation. In this case, destroy the
       // block immediately without waiting for the animation.
       afterDestroy();
     } else {
-      const destroyOverlay = this.actionGroup.create(-12 + 40 * destroyPosition[0], -22 + 40 * destroyPosition[1], "destroyOverlay", "destroy1");
+      let destroyOverlay = this.actionGroup.create(-12 + 40 * destroyPosition[0], -22 + 40 * destroyPosition[1], "destroyOverlay", "destroy1");
+      if (LevelBlock.isFlat(blockType)) {
+        let cropRect = new Phaser.Rectangle(0, 0, 60, 40);
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelView.js]: nit: both of these `let`s can be `const`s

File: src/js/game/LevelMVC/LevelView.js [hunk 2]
```diff
@@ -1150,13 +1150,18 @@ module.exports = class LevelView {
       this.playExplosionAnimation(playerPosition, facing, destroyPosition, blockType, completionHandler, true, entity);
     };
 
-    if (LevelBlock.isFlat(blockType)) {
+    if (LevelBlock.skipsDestructionOverlay(blockType)) {
```
Comment (Hamms) [inline src/js/game/LevelMVC/LevelView.js]: :+1:

File: src/js/game/LevelMVC/LevelView.js (hunk 1 above)
Comment (WilliamHarmonMS) [reply src/js/game/LevelMVC/LevelView.js]: So they -can-
I was assuming that .crop() was modifying data somehow and that const would conflict with that, but that appears to -not be the case-.

Will update right now.

Comment (Hamms) [reply src/js/game/LevelMVC/LevelView.js]: `const` only limits your ability to assign a new value to the variable, not your ability to modify the already-assigned variable;

**Pull request merged**